src/Bicep.Core.UnitTests/Utils/CompilationHelper.cs

```diff
@@ -80,12 +85,15 @@ public static CompilationResult Compile(ServiceBuilder services, params (string
 
         public static CompilationResult Compile(ServiceBuilder services, IFileResolver fileResolver, IEnumerable<Uri> sourceFiles, Uri entryUri)
         {
-            services = services.WithFileResolver(fileResolver);
+            var compiler = services.WithFileResolver(fileResolver).Build().GetCompiler();
             var sourceFileDict = sourceFiles
                 .Where(x => PathHelper.HasBicepExtension(x) || PathHelper.HasArmTemplateLikeExtension(x))
                 .ToDictionary(x => x, x => fileResolver.TryRead(x).IsSuccess(out var fileContents) ? fileContents : throw new InvalidOperationException($"Failed to find file {x}"));
 
-            return Compile(services.BuildCompilation(sourceFileDict, entryUri));
+#pragma warning disable VSTHRD002
+            var compilation = compiler.CreateCompilation(entryUri, skipRestore: true).GetAwaiter().GetResult();
+#pragma warning restore VSTHRD002
```

Comment:
How hard would it be to mark this method as `async` instead?

Reply:
Do-able, but a pretty giant refactor which I didn't feel like embarking on right now (note the number of references to each function):
<img width="1113" alt="image" src="https://github.com/Azure/bicep/assets/38542602/fc1a119e-a410-49b5-a9e9-d8d9b857c79a">

In the long run, I think it probably makes sense to work towards this.